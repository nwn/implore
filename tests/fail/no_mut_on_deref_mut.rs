use implore::impl_op;

#[derive(Copy, Clone, PartialEq, Eq, Debug)]
struct X;

#[impl_op]
fn deref_mut(x: &mut X) -> X {
    X
}

#[impl_op]
fn deref_mut(x: X) -> &mut X {
    &mut X
}

#[impl_op]
fn deref_mut(x: &mut X) -> &X {
    &X
}

#[impl_op]
fn deref_mut(x: &X) -> &mut X {
    &mut X
}

fn main() {}
