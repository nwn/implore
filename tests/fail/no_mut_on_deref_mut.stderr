error: `deref_mut` must return a mutable reference type
 --> $DIR/no_mut_on_deref_mut.rs:7:28
  |
7 | fn deref_mut(x: &mut X) -> X {
  |                            ^

error: `deref_mut` must take a mutable reference
  --> $DIR/no_mut_on_deref_mut.rs:12:17
   |
12 | fn deref_mut(x: X) -> &mut X {
   |                 ^

error: `deref_mut` must return a mutable reference type
  --> $DIR/no_mut_on_deref_mut.rs:17:28
   |
17 | fn deref_mut(x: &mut X) -> &X {
   |                            ^^

error: `deref_mut` must take a mutable reference
  --> $DIR/no_mut_on_deref_mut.rs:22:17
   |
22 | fn deref_mut(x: &X) -> &mut X {
   |                 ^^
